README.md

```diff
@@ -86,15 +82,19 @@ Use just like any other twig filter.
         {{ moreMarkdown }}
     {% endapply %}
 
-### Function
+**NOTE: Filter input is sanitised automatically.**
 
-_NOTE: The function currently **DOES NOT** sanitize input, and thus could be vulnerable to XSS if user input is not pre-sanisized. If you are unable to sanitize the user data yourself, please consider using the `{% markdown %}` tag functionality instead, as this does sanitize input._
+### Function
 
 Use just like any other twig function.
 
     {{ markdown("# Some Markdown") }}
     {{ markdown(markdownVariable) }}
 
+**NOTE: The above function usage is unsafe. Function input is not automatically saintised. To sanitise this in the template, please use the escape filter like below.**
+
+    {{ markdown(markdownVariable | escape) }}
+
```

Comment:
The markdown syntax docs here state HTML can be used to cover any feature that markdown its self does not support: https://daringfireball.net/projects/markdown/syntax#html

Reply:
This would also technically mean that even a `script` tag is perfectly fine to be included in markdown.

I'm thinking it might be best to provide all of the functionality as "safe" by default, but also providing "unsafe" versions of these filters/tags/functions as well.

Reply:
Thanks for the explanation!
In that case, it could be safer to not have `is_safe` on the function (since it can't guarantee its output is safe) and instead let users call `|raw` on the output explicitly  if they are sure their input is safe (similar to what needs to be done to output a variable containing HTML):
```twig
{% set html = '<b>foo</b>' %}

{# Dev made sure the HTML is safe #}
{{ html|raw }}
{{ markdown(html)|raw }}
```
But this would be less convenient and a breaking change so I understand if you prefer to simply document that the function is unsafe.